opensearch-project/OpenSearch-Dashboards
# [Multiple Datasource] Create examples about how to consume data source components

### Description
This change adds examples about how to configure the following components from data source management plugin:
- data source selector component 
- data source view component
- data source aggregated view to list all
- data source aggregated view to list active options
- data source multi selectable component
- data source selectable component
At the same time, there is one example shows how to configure the TopNavMenu to mount the data source component

### Issues Resolved
Fixes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6275

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/17714150/b58f3a91-6b67-4a67-bd5b-27f88e2a80e0


## Testing the changes
The following was performed in the recording
1. enable data source
2. go to developer examples, and select `Multiple Data source Integration`
3. go to the new data source selector page, should see page renders, and when selecting an option, the value is updated
4. go to multi selectable page, should see page renders, and when selecting an option, the value is updated
5. go to selectable page, should see page renders, and when selecting an option, the value is updated
6. go to read only view page, should see page renders, and component is mounted
7. go to list all page, should see page renders, and component is mounted
8. go to list active page, should see page renders, and component is mounted
9. go to last page, should see page renders, and component is mounted along with save button mounted with TopNavMenu



### Check List
